atc/atccmd/command.go

```diff
@@ -687,6 +688,10 @@ func (cmd *RunCommand) constructAPIMembers(
 			httpsHandler,
 			tlsConfig,
 		)})
+
+		if cmd.TLSOnly {
+			members = cmd.filterMembersForTLSOnly(members, "web")
```

Comment:
I think it'd be cleaner to just conditionally append the non-TLS handler instead of filtering it out:

```go
if !cmd.TLSOnly {
	members = append(members, grouper.Member{
		Name: "web",
		Runner: http_server.New(
			cmd.nonTLSBindAddr(),
			httpHandler,
		),
	})
}
```

Reply:
I thought of that but I added the filtering here to ensure that the user cant get into an a weird state where they might set TLS_ONLY to true but not have set the TLS_BIND_PORT etc. It seemed safer to filter than to do a conditional. I'm curious your thoughts on this

Reply:
Ah, yeah that makes sense. How about checking for that case in `validate()` instead? We do something similar here:

https://github.com/concourse/concourse/blob/6def8a4f5fe426150a967697ef51ee5fe5011e26/atc/atccmd/command.go#L1153-L1159